analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,357 @@
+        percent_neutral = neutral / (gain + loss + neutral),
+        dominant_status = case_when(
+          percent_gain > 0.5 &
+            percent_gain > 1.5 * percent_loss ~ "gain",
+          percent_loss > 0.5 &
+            percent_loss > 1.5 * percent_gain ~ "loss",
+          percent_neutral > 0.5 &
+            percent_neutral > 1.5 * percent_loss &
+            percent_neutral > 1.5 * percent_gain ~ "neutral",
+          TRUE ~ "none"
+        )
+      )
+  } else {
+    status_count <- status_df %>%
+      count(!!region_sym, status) %>%
+      # Spread the data -- each row represents a unique cytoband/gene
+      spread(status, n) %>%
+      replace_na(list(
+        gain = 0,
+        loss = 0,
+        amplification = 0
+      )) %>%
+      group_by(!!region_sym) %>%
+      # The logic here is to define the arm status based on the majority of calls
+      # in the region -- if the number of calls for a specific status is
+      # responsible for more than 50% of the total calls in that region, then it is
+      # used to define the region's status
+      mutate(
+        percent_gain = gain / (gain + loss + amplification),
+        percent_loss = loss / (gain + loss + amplification),
+        percent_amplification = amplification / (gain + loss + amplification),
+        dominant_status = case_when(
+          percent_gain > 0.5 &
+            percent_gain > 1.5 * percent_loss ~ "gain",
+          percent_loss > 0.5 &
+            percent_loss > 1.5 * percent_gain ~ "loss",
+          percent_amplification > 0.5 &
+            percent_amplification > 1.5 * percent_loss &
+            percent_amplification > 1.5 * percent_gain ~ "amplification",
+          TRUE ~ "neutral"
+        )
+      )
+  }
+
+  return(status_count)
+}
+
+plot_dominant_status_calls <- function(status_count_df,
+                                       status_variable) {
+  # Given a data.frame with the percentage values for each unique status in
+  # each unique region (arm/cytoband/gene) and the dominant status for that region,
+  # plot the dominant status call on the x-axis with the percentage values for a
+  # specified status variable on the y-axis.
+  #
+  # Args:
+  #   status_count_df: data.frame with percentage values for each unique status
+  #                    in each unique region (arm/cytoband/gene) and the
+  #                    dominant status for that region
+  #   status_variable: string of the column/status that contains the percentage
+  #                    values that should be plotted on the y-axis
+  # Return:
+  #   status_plot: plot representing the dominant status call on the x-axis and
+  #                the specified status coverage percentage values on the y-axis
+
+  # Make variable a symbol for plotting
+  status_sym <- rlang::sym(status_variable)
+
+  status_plot <- ggplot2::ggplot(
+    status_count_df,
+    ggplot2::aes(
+      x = dominant_status,
+      y = !!status_sym
+    )
+  ) +
+    ggplot2::geom_point() +
```

Comment:
Can you make this a jitter or a sina plot so the points get a separated from each other.